HISTORY.md

```diff
@@ -22,6 +22,7 @@
 * DB identity (`db_id`) and DB session identity (`db_session_id`) are added to table properties and stored in SST files. SST files generated from SstFileWriter and Repairer have DB identity “SST Writer” and “DB Repairer”, respectively. Their DB session IDs are generated in the same way as `DB::GetDbSessionId`. The session ID for SstFileWriter (resp., Repairer) resets every time `SstFileWriter::Open` (resp., `Repairer::Run`) is called.
 * Added experimental option BlockBasedTableOptions::optimize_filters_for_memory for reducing allocated memory size of Bloom filters (~10% savings with Jemalloc) while preserving the same general accuracy. To have an effect, the option requires format_version=5 and malloc_usable_size. Enabling this option is forward and backward compatible with existing format_version=5.
 * `BackupTableNameOption BackupableDBOptions::share_files_with_checksum_naming` is added, where `BackupTableNameOption` is an `enum` type with two enumerators `kChecksumAndFileSize` and `kChecksumAndFileSize`. By default, `BackupTableNameOption BackupableDBOptions::share_files_with_checksum_naming` is set to `kChecksumAndDbSessionId`. In this default case, backup table filenames are of the form `<file_number>_<crc32c>_<db_session_id>.sst` as opposed to `<file_number>_<crc32c>_<file_size>.sst`. The new default behavior fixes the backup file name collision problem, which might be possible at large scale, but the option `kChecksumAndFileSize` is added to allow use of old naming in case it is needed. This default behavior change is not an upgrade issue, because previous versions of RocksDB can read, restore, and delete backups using new names, and it's OK for a backup directory to use a mixture of table file naming schemes. Note that `share_files_with_checksum_naming` comes into effect only when both `share_files_with_checksum` and `share_table_files` are true.
+* A new option `std::shared_ptr<FileChecksumGenFactory> backup_checksum_gen_factory` is added to `BackupableDBOptions`. The default value for this option is `nullptr`. If this option is null, the default backup engine checksum function (crc32c) will be used for creating, verifying, or restoring backups. If it is not null, a custom checksum function will be used. When creating a new backup with non-null `backup_checksum_gen_factory`, the checksum function to be used is the one returned by `backup_checksum_gen_factory->CreateFileChecksumGenerator(FileChecksumGenContext)` with `FileChecksumGenContext::requested_function_name` being empty. When verifying or restoring a backup that was created using a custom checksum function, `Status::NotSupported` will be returned if no checksum factory is provided, and `Status::NotFound` will be returned if the name of the custom checksum function cannot be found in the checksum factory passed in by `backup_checksum_gen_factory`.
```

Comment:
Can you make this sentence more clear "When creating a new backup with non-null `backup_checksum_gen_factory`, the checksum function to be used is the one returned by `backup_checksum_gen_factory->CreateFileChecksumGenerator(FileChecksumGenContext)` with `FileChecksumGenContext::requested_function_name` being empty." It is hard to follow.